src/SAWScript/CrucibleOverride.hs

```diff
@@ -96,12 +100,20 @@ import           SAWScript.Utils (handleException)
 -- the Crucible simulation in order to compute the variable substitution
 -- and side-conditions needed to proceed.
 newtype OverrideMatcher arch mode a =
-  OM (StateT (OverrideState arch) (ExceptT OverrideFailure IO) a)
-  deriving (Functor, Applicative, Monad, MonadIO)
+  OM (StateT (OverrideState arch) (ExceptT (OverrideFailure arch) IO) a)
+  deriving (Generic, Generic1, Functor, Applicative, Monad, MonadIO)
+
+instance Wrapped (OverrideMatcher arch mode a) where
+
+instance MonadState (OverrideState arch) (OverrideMatcher arch mode) where
```

Comment:
This should actually just be done with `GeneralizedNewtypeDeriving` + `StandaloneDeriving`